### PR TITLE
refactor: layer0 to edgio

### PR DESCRIPTION
**Pull request by rishi-raj-jain**
### ❓ Type of change

- [x] 📖 Documentation (updates to the documentation or readme)

### CODE REVIEW

**Comment by rishi-raj-jain** [comment]
It's a WIP.

**Comment by danielroe** [comment]
Is this still WIP?

**Comment by rishi-raj-jain** [comment]
@danielroe Yeah, I'm on my way to create a connector with Edgio, that'd boil all of this down to:

```
npm i -g @edgio/cli@latest

edgio init
edgio build && edgio deploy
```

**Comment by rishi-raj-jain** [comment]
Because as of now what's mentioned there fails to work

**Comment by danielroe** [comment]
Converting to draft for now - just let me know whenever we're ready to go.

**Comment by pi0** [comment]
Congrats on rebranding @rishi-raj-jain.

While you are working at this, would you please also rename [layer0.ts](https://github.com/unjs/nitro/blob/main/src/presets/layer0.ts) to `edgeio.ts`? 

We can add `export const layer0 = edgeio` to keep backward compatibility.

Also would be nice that also in the docs we mention about former name and renaming (of course up to your marketing strategy)

**Comment by rishi-raj-jain** [comment]
@danielroe @pi0 Had a doubt, if I follow https://nitro.unjs.io/guide/introduction/getting-started, then following  https://nitro.unjs.io/deploy/providers/azure#local-preview leads to an error as there's no build command as there's no package.json. Is this a known area of friction? As a new user (literally which I am), I'm unable to find a package.json to copy from the site, or how to get the commands running with NITRO_PRESET. I believe this is an issue with all the guides. Hope this make sense.

**Comment by rishi-raj-jain** [comment]
Also, I got Nitropack integration with Edgio released: https://docs.edg.io/guides/changelog#:~:text=Implement%20Nitropack%20Connector%20(%231589)%20%40rishi%2Draj%2Djain

**Comment by rishi-raj-jain** [comment]
I've updated the edgio.ts file, but may you let me know how shall I test it (that NITRO_PRESET=edgio yarn build) works as expected (per the question above)?

**Comment by pi0** [comment]
Hi dear @rishi-raj-jain thanks for updates. Let me locally test it. 

Normally, nitro has presets that _generates_ the expected files including ideal `package.json` into the `.output` directory that edgio can pick and deploy it.

**Comment by rishi-raj-jain** [comment]
@pi0 Did that turn out well?

**Comment by rishi-raj-jain** [comment]
@pi0 Did that turn out well?

**Comment by rishi-raj-jain** [comment]
@pi0 Did that turn out well?

**Comment by rishi-raj-jain** [comment]
@pi0 Did that turn out well?

**Comment by rishi-raj-jain** [comment]
@danielroe I think it's ready to be reviewed.

File: src/presets/edgio.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+async function writeFile (path: string, contents: string) {
+  await fsp.mkdir(dirname(path), { recursive: true })
+  await fsp.writeFile(path, contents, 'utf-8')
+}
+
+// Edgio router (.output/routes.js)
+function routesTemplate () {
+  return `
+import { Router } from '@edgio/core/router'
+import { nitropackRoutes } from '@edgio/nitropack'
```
Comment (pi0) [inline src/presets/edgio.ts:52]: We can generate `nitropackRoutes` in the output directory directly. [implementation](https://unpkg.com/browse/@edgio/nitropack@6.0.5/nitropackRoutes.js. `.output/public` for example is configurable. And for default-app import, we need to use same (generated) file.

File: docs/content/2.deploy/providers/edgio.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+## Testing locally
+
+You can use Edgio to test your app's developement experience locally:
+
+```bash
+edgio dev
+```
+
+Under the hood, `edgio dev` runs `npx nitropack dev` on the port 3001.
```
Comment (pi0) [inline docs/content/2.deploy/providers/edgio.md:34]: I prefer to advice `nuxt`/`nitro` CLI primarily for development for consistency (unless there is a specific feature you want in dev mode, would be happy to discuss how we can implement it as cli/framework extension)

File: docs/content/2.deploy/providers/edgio.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+## Install the Edgio CLI
+
+```bash
+npm i -g @edgio/cli
+```
+
+## Initialize your project for use with Edgio
+
+```bash
+edgio init --connector=@edgio/nitropack
```
Comment (pi0) [inline docs/content/2.deploy/providers/edgio.md:23]: We might instead suggest to use `preset: `edgio` in the config file.

File: src/presets/index.ts [hunk 4]
```diff
@@ -7,7 +7,7 @@ export * from './deno'
 export * from './digital-ocean'
 export * from './firebase'
 export * from './heroku'
-export * from './layer0'
+export * from './edgio'
```
Comment (pi0) [inline src/presets/index.ts:10]: This is a breaking change. I hope it is fine for your current customers...

**Review by pi0 — CHANGES_REQUESTED** [review]

File: src/presets/index.ts (hunk 4 above)
Comment (manniL) [reply src/presets/index.ts:10]: Maybe it'd be good to keep it as an alias?

Comment (pi0) [reply src/presets/index.ts:10]: Yes

**Comment by rishi-raj-jain** [comment]
Closed against https://github.com/unjs/nitro/pull/822

**Pull request closed without merging**